core/src/test/java/org/wso2/carbon/transports/TransportManagerTest.java

```diff
@@ -0,0 +1,116 @@
+        carbonTransport2 = new CustomCarbonTransport("dummyId2");
+        transportManager.registerTransport(carbonTransport);
+        transportManager.registerTransport(carbonTransport2);
+    }
+
+    @Test
+    public void testUnsuccessfullStartTransport() {
+        try {
+            transportManager.startTransport("wrongId");
+        } catch (IllegalArgumentException e) {
+            String exceptionMessage = "wrongId not found";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+
+    @Test(dependsOnMethods = {"testUnsuccessfullStartTransport"})
+    public void testSuccessfullStartTransport() {
+        try {
+            transportManager.startTransport("dummyId");
+        } catch (IllegalArgumentException e) {
+            Assert.assertFalse(true);
+        }
+        Assert.assertTrue(true);
+    }
+
+    @Test(dependsOnMethods = {"testSuccessfullStartTransport"})
+    public void testUnsuccessfullStopTransport() {
+        try {
+            transportManager.stopTransport("wrongId");
+        } catch (IllegalArgumentException e) {
+            String exceptionMessage = "wrongId not found";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+    @Test(dependsOnMethods = {"testUnsuccessfullStopTransport"})
+    public void testSuccessfullStopTransport() {
+        try {
+            transportManager.stopTransport("dummyId");
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(false);
+        }
+        Assert.assertTrue(true);
+    }
+
+    @Test(dependsOnMethods = {"testSuccessfullStopTransport"})
+    public void testUnregisterTransport() {
+        try {
+            transportManager.unregisterTransport(carbonTransport);
+            transportManager.stopTransport(carbonTransport.getId());
+        } catch (IllegalArgumentException e) {
+            String exceptionMessage = "dummyId not found";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+    @Test(dependsOnMethods = {"testUnregisterTransport"})
+    public void testUnsuccessfulBeginMaintenance() {
+        try {
+            transportManager.registerTransport(carbonTransport2);
+            transportManager.beginMaintenance();
+        } catch (IllegalStateException e) {
+            String exceptionMessage = "Cannot put transport dummyId2 into maintenance. Current state: UNINITIALIZED";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+    @Test(dependsOnMethods = {"testUnsuccessfulBeginMaintenance"})
+    public void testSuccessfulBeginMaintenance() {
+        try {
+            transportManager.registerTransport(carbonTransport2);
+            transportManager.startTransport(carbonTransport2.getId());
+            transportManager.beginMaintenance();
+        } catch (IllegalStateException e) {
+            Assert.assertTrue(false);
+        }
+        Assert.assertTrue(true);
+    }
+
+}
```

Comment:
Check if it is possible to use expectedexception where test cases throws an exception as the result